esm/html/document.js

```diff
@@ -102,3 +103,53 @@ export class HTMLDocument extends Document {
     return element;
   }
 }
+
+/** @typedef {{ [WINDOW] : Object }} Context */
+/**
+ * @constructor
+ * @param {Context} context
+ */
+export class HTMLDocument extends _HTMLDocument {
+
+  constructor (context = {}) {
+    super();
+    const useState = (target, name) => {
+      return ((_0, _1={}) => {
+        const g = (name in _1) ? _1 : _0;
+        return {
+          get: () => g[name],
+          set: (v) => (g[name] = v) || true,
+        }
+      })(target, context[WINDOW]);
+    };
+    this[WINDOW] = {
+      set: (target, name, value) => {
+        return useState(target, name).set(value);
+      },
+      get: (target, name) => {
+        if (name === 'window') {
+          return new Proxy(target.window, this[WINDOW]);
+        }
+        return useState(target, name).get();
+      }
+    }
+  }
+
+  get defaultView() {
+    return new Proxy(super.defaultView, this[WINDOW])
+  }
+
+  get location() {
+    return this.defaultView.location;
+  }
```

Comment:
A special "location" property is shared between `window` and `document`.

Reply:
if this is how the standard works I'm OK with it, if not, we need to think about a better approach that is more flexible around shared things

Reply:
A quick [code sandbox](https://codesandbox.io/s/window-and-document-0pngb?file=/src/index.ts) identifies only "location" as the only non-null shared property that linkedom doesn't yet cover. However, I've recently noticed 89 properties like "onload" that default to `null` in both `document` and `window`.

For completeness, we should check [the HTML standard](https://html.spec.whatwg.org/) whether those 89 properties identify the same objects in both [window](https://html.spec.whatwg.org/multipage/window-object.html#window) and [document](https://html.spec.whatwg.org/multipage/dom.html#document).